mypy_django_plugin/transformers/models.py

```diff
@@ -658,16 +658,27 @@ def run(self) -> None:
             # TODO: Create abstract through models?
             return
 
-        # Start out by prefetching a couple of dependencies needed to be able to declare any
-        # new, implicit, through model class.
-        model_base = self.lookup_typeinfo(fullnames.MODEL_CLASS_FULLNAME)
-        fk_field = self.lookup_typeinfo(fullnames.FOREIGN_KEY_FULLNAME)
-        manager_info = self.lookup_typeinfo(fullnames.MANAGER_CLASS_FULLNAME)
-        if model_base is None or fk_field is None or manager_info is None:
-            raise helpers.IncompleteDefnException()
-
-        from_pk = self.get_pk_instance(self.model_classdef.info)
-        fk_set_type, fk_get_type = get_field_descriptor_types(fk_field, is_set_nullable=False, is_get_nullable=False)
+        if self.model_classdef.fullname == fullnames.USER_MODEL_FULLNAME:
+            permission_typeinfo = self.lookup_typeinfo_or_incomplete_defn_error(fullnames.PERMISSION_MODEL_FULLNAME)
+            self.create_through_table_class(
+                field_name="user_permissions",
+                model_name="User_user_permissions",
+                model_fullname="django.contrib.auth.models.User_user_permissions",
+                m2m_args=M2MArguments(
+                    to=M2MTo(arg=self.model_classdef, model=Instance(permission_typeinfo, []), self=False), through=None
+                ),
+            )
+            group_typeinfo = self.lookup_typeinfo_or_incomplete_defn_error(fullnames.GROUP_MODEL_FULLNAME)
+            self.create_through_table_class(
+                field_name="groups",
+                model_name="User_groups",
+                model_fullname="django.contrib.auth.models.User_groups",
+                m2m_args=M2MArguments(
+                    to=M2MTo(arg=self.model_classdef, model=Instance(group_typeinfo, []), self=False),
+                    through=None,
+                ),
+            )
+            return
```

Comment:
Hm, I don't think we need these through models in the plugin. I think we can add them explicitly in `django/contrib/auth/models.py` with a `@type_check_only` decorator

And then also add the following:

```diff
class User(AbstractUser):
    id: models.AutoField[str | int | Combinable | None, int]
    pk: models.AutoField[str | int | Combinable | None, int]
+   groups: models.ManyToManyField[Group, User_groups]
+   user_permissions: models.ManyToManyField[Permission, User_user_permissions]
```